File: compatibility-tests/basic/gradle-java/src/test/java/CollectionTest.java
import org.tabletest.junit.TableTest;
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.List;

public class CollectionTest {
    @TableTest("""
        numbers   | size?
        [1, 2, 3] | 3
        []        | 0
        [5]       | 1
        """)
    void list_parameter(List<String> numbers, int size) {
        assertEquals(size, numbers.size());
    }
}

File: compatibility-tests/basic/maven-kotlin/src/test/kotlin/CollectionTest.kt
import org.tabletest.junit.TableTest
import kotlin.test.assertEquals

class CollectionTest {
    @TableTest("""
        numbers   | size?
        [1, 2, 3] | 3
        []        | 0
        [5]       | 1
        """)
    fun list_parameter(numbers: List<String>, size: Int) {
        assertEquals(size, numbers.size)
    }
}
